### DISCUSSION TITLE
How do stream limits work?

### DISCUSSION THREAD

**Discussion by jeremyong** [microsoft/msquic#1441] [Q&A] [answered]
I can't tell if this is a bug in my application or not, but after the max stream limit has been reached (the value of `PeerBidiStreamCount`), new streams cannot be opened even if old streams are shutdown/closed. I've verified that stream shutdown events are posted on both sides (either as `QUIC_STREAM_EVENT_PEER_SEND_SHUTDOWN` or `QUIC_STREAM_EVENT_SHUTDOWN_COMPLETE` depending on which side initiated the shutdown).

Is the limit a hard limit for the lifetime of the connection, or am I doing something wrong somewhere?

Thanks

**Comment by nibanks** [comment, marked as answer]
`PereBidiStreamCount` is not a hard limit. It automatically allows for more streams to be created once the old ones are shutdown in both directions (not just the send direction). The stream has to be shutdown completely by both sides before the transport considers it gone, and allows for a new one to be created.

**Reply by jeremyong** [reply]
Ok thanks it was the shutdown on both sides that I was missing since the streams are bidirectional by default.
